videostore/utils.go

```diff
@@ -229,41 +263,85 @@ func formatDateTimeToString(dateTime time.Time) string {
 
 // matchStorageToRange returns a list of files that fall within the provided time range.
 // Includes trimming video files to the time range if they overlap.
-func matchStorageToRange(files []string, start, end time.Time, logger logging.Logger) []string {
+// The matcher assumes that the input files list is sorted by start time and the underylying
+// video segments do not overlap.
+func matchStorageToRange(files []fileWithDate, start, end time.Time, logger logging.Logger) []string {
```

Comment:
do we need to check that start < end?

Reply:
This is already validated in the do-command validation step.
https://github.com/viam-modules/video-store/blob/main/videostore/videostore.go#L126-L131